### PR TITLE
minor: Remove closed issue reference

### PR DESCRIPTION
**Pull request by suniti0804**
As the [Issue #13544](https://github.com/checkstyle/checkstyle/issues/13544) has been marked closed so we need to remove it's reference from the files to avoid build failures.

Build failure discovered in - https://github.com/checkstyle/checkstyle/actions/runs/6941703823/job/18883087013?pr=13951

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java [hunk 1]
```diff
@@ -112,7 +111,7 @@ public Violation(int lineNo,
             this.args = null;
         }
         else {
-            this.args = UnmodifiableCollectionUtil.copyOfArray(args, args.length);
+            this.args = Arrays.copyOf(args, args.length);
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java:114]: Please revert this change

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitCyclesCheckTest.java [hunk 2]
```diff
@@ -30,7 +29,6 @@
 import com.tngtech.archunit.lang.ArchRule;
 import com.tngtech.archunit.library.freeze.FreezingArchRule;
 
-@Disabled("until https://github.com/checkstyle/checkstyle/issues/13544")
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitCyclesCheckTest.java:33]: We can now reference https://github.com/checkstyle/checkstyle/issues/14053 in disabled annotation, please do this for all.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java (hunk 1 above)
Comment (suniti0804) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java]: done

**Review by nrmancuso — APPROVED** [review]
Ok to merge as CI passes, @suniti0804 thank you for noticing and sending a fix :)

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitCyclesCheckTest.java (hunk 2 above)
Comment (suniti0804) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitCyclesCheckTest.java:33]: done

**Comment by suniti0804** [comment]
> Ok to merge as CI passes, @suniti0804 thank you for noticing and sending a fix :)

Happy to contribute :)
